src/Controller/Controller.php

```diff
@@ -178,7 +179,7 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * @param \Cake\Event\EventManager|null $eventManager The event manager. Defaults to a new instance.
      * @param \Cake\Controller\ComponentRegistry|null $components The component registry. Defaults to a new instance.
      */
-    public function __construct(ServerRequest $request = null, Response $response = null, $name = null, $eventManager = null, $components = null)
+    public function __construct(ServerRequest $request = null, Response $response = null, ?string $name = null, ?\Cake\Event\EventManager $eventManager = null, ?\Cake\Controller\ComponentRegistry $components = null)
```

Comment:
Need to add `use` statements instead of using FQCNs. Also `EventManager` needs to be changed to `EventManagerInterface`.